src/components/Authorization.js

```diff
@@ -0,0 +1,198 @@
+    static entryPath = '/wall'
+
+    // Where it all begins.
+    static rootPath = '/'
+
+    // And where everything will end.
+    static logoutPath = '/logout'
+
+    // Authorization context consumer.
+    static Consumer = AuthorizationContext.Consumer
+
+    storage = createStoreAccess('auth')
+
+    state = {
+        // State of the context passed to consumers.
+        context: {
+            authorized: false,
+            token: null,
+            userId: null,
+            userName: null,
+        }
+    }
+
+    get currentPath() {
+        return this.props.location.pathname
+    }
+
+    componentWillMount() {
+        const paramToken = this.getTokenFromParams()
+        const currentToken = this.getTokenFromStorage()
+
+        if (currentToken) {
+            this.startFromStoredData()
+        } else if (this.isAtLogin() && paramToken) {
+            this.startSessionFromAPIInfo()
+        }
+    }
+
+    componentWillReceiveProps({location}) {
+        if (location.pathname === Authorization.logoutPath) {
+            this.endSession()
+        }
+    }
+
+    render() {
+        return (
+            <AuthorizationContext.Provider value={this.state.context}>
+                {this.props.children}
+            </AuthorizationContext.Provider>
+        )
+    }
+
+    startFromStoredData() {
+        const currentToken = this.getTokenFromStorage()
+        const storedUser = this.getUserFromStorage()
+        try {
+            const user = JSON.parse(storedUser)
+            if (this.isUserValid(user)) {
+                this.setCredentials({token: currentToken, ...user})
+                if (this.isAtLogin() || this.isAtRoot()) {
+                    this.navigateIn()
+                }
+            } else {
+                this.handleInvalidUserData()
+            }
+        } catch (error) {
+            this.handleInvalidUserData()
+        }
+    }
+
+    startSessionFromAPIInfo() {
+        if (!this.areParamsValid()) {
+            return this.handleInvalidUserData()
+        }
+        const token = this.getTokenFromParams()
+        const user = this.getUserFromParams()
+
+        try {
+            this.storeSessionInfo({token, ...user})
+        } catch (error) {
+            this.handleInvalidUserData()
+        }
+
+        this.setCredentials({token, ...user})
+        this.navigateIn()
+    }
+
+    endSession() {
+        this.clearStoredData()
+        this.unsetCredentials()
+        this.navigateToRoot()
+    }
+
+    setCredentials({token, name, id}) {
+        this.setState(state => ({
+            context: {
+                ...state.context,
+                token,
+                authorized: Boolean(token),
+                userName: name,
+                userId: id,
+            }
+        }))
+    }
+
+    unsetCredentials() {
+        this.setCredentials({token: null, name: null, id: null})
+    }
+
+    isUserValid({name, id} = {}) {
+        return Boolean(name && id)
```

Comment:
I could be more certain with using: `Boolean(name) && Boolean(id)`

Reply:
Certain of what?